tests/test_health.py

```diff
@@ -949,3 +949,39 @@ def test_positive_available_space_postgresql12(ansible_module):
 
     :CaseImportance: High
     """
+
+
+def test_positive_check_duplicate_permissions(ansible_module, setup_duplicate_permissions):
+    """Verify duplicate-permissions check
+
+    :id: 192a5943-43cc-4f89-9949-893c8011831a
+
+    :setup:
+        1. foreman-maintain should be installed.
+        2. Setup duplicate permissions in the database
+
+    :steps:
+        1. foreman-maintain health check --label duplicate-permissions
+
+    :expectedresults: Verify if the check passes and removes duplicate permissions.
+
+    :CaseImportance: High
+
+    :BZ: 1849110, 1884024
+
+    """
+    # Verify if check failed because of duplicate permissions
+    contacted = ansible_module.command(
+        Health.check(["--label", "duplicate-permissions", "--assumeyes"])
+    )
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert "FAIL" in result["stdout"]
+        assert "Remove duplicate permissions from database" in result["stdout"]
+        assert result["rc"] == 0
+    # Verify the check passed
+    contacted = ansible_module.command(Health.check(["--label", "duplicate-permissions"]))
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert "FAIL" not in result["stdout"]
+        assert result["rc"] == 0
```

Comment:
Non blocking comments.

I don't know what exactly what `result` contains, just looking on the code, but if rc == 0, I don't think asserting on "FAIL" is much helpful. 
Yes, If rc is non zero then such assertion does help.

Just a helpful tip, I'd recommend converting either FAIL or result["stdout"] to a common upper/lower case. Obviously you can create a PR and fix it later if such assertion fails. :)

Reply:
Wont the usual 'assert result.status == 0' work in this case?

Reply:
IMHO both checks are fine here. The first run assures the `FAIL` is in stdout (because of duplicate permissions) and that the health-check fixes the isssue (`--assumeyes`) so the RC is zero. The second run assures there is no duplicate anymore - so no `FAIL` in stdout this time and RC still 0.
I agree one check would work too but this way I feel the test purpose is better readable.

Reply:
Re "test purpose is better readable.", yes, like self documenting, good point.

Reply:
> Wont the usual 'assert result.status == 0' work in this case?

@swadeley , `result` here is a dictionary, hence it wont work :)

Reply:
Thanks for your input @akhil-jha, For this test case I would like to keep both the assertions as Vladimir mentioned first check ensures FAIL in stdout and rc=0 and the second check ensures duplicate permissions are removed :)